src/controller.cpp

```diff
@@ -342,16 +345,19 @@ void Controller::logout() {
 
   m_nextStep = Disconnect;
 
-  if (m_state == StateOn) {
+  if (m_state == StateOn || m_state == StateOnPartial) {
     deactivate();
     return;
   }
 }
 
-void Controller::activateInternal(DNSPortPolicy dnsPort,
-                                  ServerSelectionPolicy serverSelectionPolicy) {
+void Controller::activateInternal(
+    DNSPortPolicy dnsPort,
+    ServerSelectionPolicy serverSelectionPolicy = RandomizeServerSelection,
+    ActivationPrincipal initiator = ClientUser) {
   logger.debug() << "Activation internal";
   Q_ASSERT(m_impl);
+  m_iniator = initiator;
```

Comment:
Should this be m_initiator?

Reply:
We need to save the initiator passed to activateInternal, it's not getting set anywhere else 😓 (except for resetting it on deactivation)

Reply:
Or did you point out the obvious typo? Hahaha. yes.